frontend/src/metabase/dashboard/components/DashboardHeader/buttons/AddFilterParameterButton/AddFilterParameterButton.unit.spec.tsx

```diff
@@ -0,0 +1,95 @@
+import { screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+
+import { renderWithProviders } from "__support__/ui";
+import {
+  createMockDashboardState,
+  createMockState,
+} from "metabase-types/store/mocks";
+
+import { AddFilterParameterButton } from "./AddFilterParameterButton";
+
+const setup = ({ isAddParameterPopoverOpen = false } = {}) => {
+  const state = createMockState({
+    dashboard: createMockDashboardState({
+      isAddParameterPopoverOpen,
+    }),
+  });
+
+  return renderWithProviders(<AddFilterParameterButton />, {
+    storeInitialState: state,
+  });
+};
+
+describe("AddFilterParameterButton", () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("should render the button with correct icon and tooltip", () => {
+    setup();
+    const button = screen.getByLabelText("Add a filter");
+    expect(button).toBeInTheDocument();
+    expect(screen.getByLabelText("filter icon")).toBeInTheDocument();
+  });
+
+  describe("when popover is open based on state", () => {
+    it("should render the popover when isAddParameterPopoverOpen is true", () => {
+      setup({ isAddParameterPopoverOpen: true });
+      expect(
+        screen.getByTestId("add-filter-parameter-dropdown"),
+      ).toBeInTheDocument();
+    });
+    it("should not render the popover when isAddParameterPopoverOpen is false", () => {
+      setup({ isAddParameterPopoverOpen: false });
+      expect(
+        screen.queryByTestId("add-filter-parameter-dropdown"),
+      ).not.toBeInTheDocument();
+    });
+  });
```

Comment:
I'm wondering if we need these two tests, they're a bit coupled with the implementation.
I think we could even remove `isAddParameterPopoverOpen` in the other tests and just open the button by clicking on it as it's just a line, what do you think? It'd be closer to what the end user is actually seeing

Reply:
yeah, I'm a bit torn on this one. On one hand, it's a popover and we should test the open/close behaviors without the actual implementation detail of where that boolean exists. On the other hand, we are basing things on the state of the redux store so we should make sure it works.


which brings the other question - do we really need this in the store?

Reply:
that refactor could be done in another PR - I think we could maybe solve this issue first since that's the point of this PR

Reply:
I think that's a reason to not test implementation details right? If in the tests we click on the button to open the popover, the tests will work as expected even if we move the logic out of redux right?

Reply:
ah we can't move it out of redux, we have some logic in `frontend/src/metabase/parameters/components/ParameterLinkedFilters/ParameterLinkedFilters.tsx` that uses the redux store

Reply:
but yes you're right

Reply:
> i guess?

we're in the same boat then 😄